PWGHF/TableProducer/candidateCreatorBplus.cxx

```diff
@@ -42,7 +43,8 @@ using namespace o2::aod::hf_cand_bplus;
 
 /// Reconstruction of B± → D0bar(D0) π± → (K± π∓) π±
 struct HfCandidateCreatorBplus {
-  Produces<aod::HfCandBplusBase> rowCandidateBase;
+  Produces<aod::HfCandBplusBase> rowCandidateBase;     // table defined in CandidateReconstructionTables.h
+  Produces<aod::HfCandBplusProngs> rowCandidateProngs; // table defined in CandidateReconstructionTables.h
```

Comment:
The initial motivation for this "splitting" of the candidate table into 2 tables (i.e. extracting the prongs indices in a new table) was to have a common table (the Base one) for both Standard and Reduced workflows, while the table with prongs can be different for Standard and Reduced workflows (as the table the index columns are pointing to different tables).

Actually, thanks to [this kind of line](https://github.com/AliceO2Group/O2Physics/pull/3401/files#diff-bc149e5241d965436914a3a7c144b71bcceb435b5bd18cf7887925e11151355dR752), we subscribe to the joined tables in selectors and analysis tasks.

Reply:
If I get it properly, the thing is that in this `candidateCreatorBplus.cxx` the candidate D0 daughter can be queried only from the `aod::HfCand2Prong` when  `candsDPerCollision` is defined. Therefore, one should also save the `aod::HfCand2Prong` but this is not what you want (by definition, otherwise you would not have started this job). In addition, also the `Preslice` is t=done referring to the tables either of the normal collisions (here) or the reduced one (in the `candidateCreatorBplusReduced`), introducing the same storing issue als ofor the collisions.
Am I right?

Reply:
Yes but that's not the whole story (sadly for us ^^ that's why we had to fix the current version).

There are 2 different points:
1. As you said, we can't store the whole candidate tables (B0, Bplus) so we needed to define our own with minimal number of columns.
*Doing so, everything worked (the current version on the master branch is functional but only locally, without using `o2-aod-merger`).*
2. The second point is tricky. With the current version on the master branch, our reduced AODs for B0, Bplus candidates contain index columns pointing towards Standard 2/3-prong candidates tables. As we don't store these Standard tables, the `o2-aod-merger` was looking for indices to tables that are not there. The solution we went with consists in simply extracting these *culprit* column indices in dedicated tables and create a Standard version of these tables (with columns pointing to Standard tables) and a Reduced version of these tables (with columns pointing to Reduced tables). This fix contains the whole "self-contained" aspect of this new version of the workflow.

Reply:
What you say reflects (and explains better wrt my comment) what I understood. Thanks again.